sale_quotation_auto_confirm/models/sale_order.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, timedelta
+
+from odoo import models, fields, api
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def action_quotation_auto_confirm(self):
+        threshold_date = (datetime.today() + timedelta(days=7)).strftime(
+            '%Y-%m-%d %H:%M:%S')
+        pending_quotations = self.search([
+            ('state', 'in', ('sent', 'draft')),
+            ('team_id', '=', self.env.ref(
+                'sales_team.team_sales_department').id),
+            '|',
+            ('requested_date', '<=', threshold_date),
+            ('requested_date', '=', False),
+            ('commitment_date', '<=', threshold_date),
```

Comment:
@yostashiro Is the domain here fine?
i.e. Sales Orders that are having 
( status in 'sent' or 'draft' ) and 
( sales team is 'Direct Sales' ) and 
(
	( Requested Date is after 7 days or more than the current date ) or 
	(( Requested Date is empty ) and ( Commitment Date is after 7 days or more than the current date))
)